File: stacks_and_queues/make_the_string_great.java
public String makeGood(String s)
{
    // Use stack to store the visited characters
    Stack<Character> stack = new Stack<>();

    // Iterate over s
    for (char currChar : s.toCharArray())
    {
        // If the current character makes a pair with the last char in the stack
        // remove both of them. Else, add current char to stack
        if (!stack.isEmpty() && Math.abs(stack.lastElement() - currChar) == 32)
        {
            stack.pop();
        }
        else
        {
            stack.add(currChar);
        }
    }

    // Return string concatenated by all chars left in stack
    StringBuilder ans = new StringBuilder();
    for (char currChar : stack)
    {
        ans.append(currChar);
    }
    return ans.toString();
}

File: stacks_and_queues/make_the_string_great.py
def makeGood(s: str) -> str:
    # Use stack to store the visited chars
    stack = []

    # Iterate over s
    for curr_char in list(s):
        # If the current char makes a pair with the last char in the stack then
        # remove both of them. Else, add current char to stack
        if stack and abs(ord(stack[-1]) - ord(curr_char)) == 32:
            stack.pop()
        else:
            stack.append(curr_char)
    
    # Returns string concatenated by all chars left in the stack
    return "".join(stack)